### ISSUE TITLE
Docker image won't stop restarting

### ISSUE THREAD

**Issue by bison92** [Azure/Azurite#2438]
### Which service(blob, file, queue, table) does this issue concern?

Docker version of Azurite

### Which version of the Azurite was used?

latest- just pulled from registry. mcr.microsoft.com/azure-storage/azurite                      latest    d9dc9980c4d0 

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)

mcr.microsoft.com

### What's the Node.js version?

docker.

### What problem was encountered?

Container won't run and only restarts outputting this line: 

`Exit due to unhandled error: unsupported`

### Steps to reproduce the issue?

I'm running the image with the following docker-compose : 

```yaml

  azuretablestorage:
    image: "mcr.microsoft.com/azure-storage/azurite"
    restart: always
    command: "azurite --blobPort ${AZURITE_PORT_PREFIX}00 --queuePort ${AZURITE_PORT_PREFIX}01 --tablePort ${AZURITE_PORT_PREFIX}02 --blobHost 0.0.0.0 --tableHost 0.0.0.0 --queueHost 0.0.0.0 --cert /https/aspnetapp.pfx --pwd 123123"
    volumes:
      - ${HTTPS_LOCATION}/servicehub.pfx:/https/aspnetapp.pfx:ro
    ports:
      - "${AZURITE_PORT_PREFIX}00:${AZURITE_PORT_PREFIX}00"
      - "${AZURITE_PORT_PREFIX}01:${AZURITE_PORT_PREFIX}01"
      - "${AZURITE_PORT_PREFIX}02:${AZURITE_PORT_PREFIX}02"
    networks:
      - netBackEnd
```

I tried adding a volume and the -d parameter, pointing to a file there, but it still fails with the same error and no debug file is created.

### Have you found a mitigation/solution?

No solution yet, except to use some other flavor outside docker.

**Comment by blueww** [comment]
@bison92 

I can't repro this issue.
I tested with following docker-compose.yml file, and it works well on my machine with latest Azurite docker image.
```
services:
  azuretablestorage:
    image: "mcr.microsoft.com/azure-storage/azurite"
    restart: always
    command: "azurite --blobPort 10000 --queuePort 10001 --tablePort 10002 --blobHost 0.0.0.0 --tableHost 0.0.0.0 --queueHost 0.0.0.0 --cert /https/cert123456.pfx --pwd 123456"
    volumes:
      - C:\temp\azurite\cert123456.pfx:/https/cert123456.pfx:ro
    ports:
      - "10000:10000"
      - "10001:10001"
      - "10002:10002"
 ```
Please note,
1. I added "services:" in the first line, or the `docker compose up` command will fail on my machine with " (root) Additional property azuretablestorage is not allowed".
2. I removed the networks netBackEnd, since netBackEnd is not on my machine
3. I don't use the parameters to input the port and file location. 

However, all these difference are not related with Azurite itself. Please check if you have any wrong configuration on your machine or docker compose file.

**Comment by bison92** [comment]
Hi, it has been working before and it works for all my colleagues without issue, so I'm guessing it is not related with the actual docker compose configuration, but that's all I have :/ 

Today I've tried removing all containers, images and volumes.

`docker stop $(docker ps -q)` 

`docker rm $(docker ps -aq)`

`docker rmi $(docker images -q)`

`docker system prune`

`docker volume prune`

After doing this the output of `docker system df` was all 0s. 

Still the same situation: 

```
$ docker logs -f 5bfffbcbee2e
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
Exit due to unhandled error: unsupported
```

I also trying commenting my docker-compose service and using the code you provided (except for the certificate). 
I will try regenerating the certificate, but its working fine on the rest of containers. 

Any ideas on what else could I be trying?



What else could I try?

**Comment by blueww** [comment]
@bison92 

I am not sure why it failed on your machine. 
However, as it work for all your colleagues , and also work for me, this looks not an Azurite issue but your local configure/environment issue.

You might can contact docker team to see if they have any idea.
And since the issue is restart, how about remove "restart: always" from yml file, and see what will happen.

**Comment by bison92** [comment]
Today I tried starting over. I deleted the current ubuntu distro from wsl, and installed the latest one 24.04. 

I installed docker 27.1.1 and configured it so I can use it from windows. 

Right after installing the latest docker I went to docker hub and took the first example on how to run the azurite container:

`docker run -p 10000:10000 -p 10001:10001 -p 10002:10002 mcr.microsoft.com/azure-storage/azurite` 

 That worked.

After finishing my setup I went to run my docker-compose project and everything was good until I had to use the azurite container defined there. To my surprise it was restarting again. 

This time the error that the container was giving was different: 

`Exit due to unhandled error: mac verify failure`

The fact that the original command without certificate worked and but with the certificate it was not working, already pointed to the cert, but the cert was good for the rest of the containers... 

Googling for this new error it looks like recent NodeJs versions (like the one the latest azurite image uses, I'm guessing) use openssl3 and have compatibility issues when trying to decrypt certificates created with previous openssl versions. 

After following the thread a bit I found https://github.com/nodejs/node/issues/40672#issuecomment-1243648223

Where they explain how to upgrade a PFX with a couple of comands:

> ```
> openssl pkcs12 -in oldPfxFile.pfx -nodes -legacy -out decryptedPfxFile.tmp
> openssl pkcs12 -in decryptedPfxFile.tmp -export -out newPfxFile.pfx
> ```

I've run them on WSL and they worked like charm, after upgrading the PFX file the issue got away and my azurite container started working.
